ReportSimulationOutput/measure.rb

```diff
@@ -872,6 +919,82 @@ def get_outputs(runner, args)
+            battery_id = elcs.additionalProperties.getFeatureAsString('HPXML_ID')
+            next unless (battery_id.is_initialized && battery_id.get == battery.id)
+
+            batt_kwh = elcs.additionalProperties.getFeatureAsDouble('UsableCapacity_kWh').get
+            batt_loss = "#{Constants.ObjectNameBatteryLossesAdjustment(elcs.name)} outvar"
+          end
+        end
+
+        resilience_frequency = 'timestep'
+        ts_per_hr = @model.getTimestep.numberOfTimestepsPerHour
+        if args[:timeseries_frequency] != 'timestep'
+          resilience_frequency = 'hourly'
+          ts_per_hr = 1
+        end
+
+        batt_soc = get_report_variable_data_timeseries(keys, vars, 1, 0, resilience_frequency)
+        batt_soc_kwh = batt_soc.map { |soc| soc - minimum_storage_state_of_charge_fraction }.map { |soc| soc * batt_kwh }
+        elec_prod = get_report_meter_data_timeseries(['ElectricityProduced:Facility'], UnitConversions.convert(1.0, 'J', 'kWh'), 0, resilience_frequency)
+        elec_stor = get_report_meter_data_timeseries(['ElectricStorage:ElectricityProduced'], UnitConversions.convert(1.0, 'J', 'kWh'), 0, resilience_frequency)
+        batt_loss = get_report_variable_data_timeseries(['EMS'], [batt_loss], UnitConversions.convert(1.0, 'J', 'kWh'), 0, resilience_frequency)
+        elec_prod = elec_prod.zip(elec_stor).map { |x, y| -1 * (x - y) }
+        elec = get_report_meter_data_timeseries(['Electricity:Facility'], UnitConversions.convert(1.0, 'J', 'kWh'), 0, resilience_frequency)
+        crit_load = elec.zip(elec_prod, batt_loss).map { |x, y, z| x + y + z }
+
+        resilience_timeseries = []
+        n_timesteps = crit_load.size
+        (0...n_timesteps).each do |init_time_step|
+          resilience_timeseries << get_resilience_timeseries(init_time_step, batt_kwh, batt_kw, batt_soc_kwh[init_time_step], crit_load, batt_roundtrip_eff, n_timesteps, ts_per_hr)
+        end
+
+        resilience.annual_output = resilience_timeseries.sum(0.0) / resilience_timeseries.size
+        resilience.timeseries_output = resilience_timeseries
+
+        # Aggregate up from hourly to the desired timeseries frequency
+        next unless ['daily', 'monthly'].include? args[:timeseries_frequency]
+
+        year = @hpxml.header.sim_calendar_year
+        sim_start_day_of_year, sim_end_day_of_year, _sim_start_hour, _sim_end_hour = get_sim_times_of_year(year)
+        n_hours_per_period = get_n_hours_per_period(args[:timeseries_frequency], sim_start_day_of_year, sim_end_day_of_year, year)
+
+        resilience_timeseries = []
+        start_hour = 0
+        n_hours_per_period.each do |n_hours|
+          resilience_timeseries << resilience.timeseries_output[start_hour..start_hour + n_hours - 1].sum(0.0) / resilience.timeseries_output[start_hour..start_hour + n_hours - 1].size
+          start_hour += n_hours
+        end
+        resilience.timeseries_output = resilience_timeseries
```

Comment:
You created a `get_n_hours_per_period` method, which is good and helps reduce the duplicated code, but there is still a lot of duplicated code between here and the emissions code below. Can we replace that method with a `rollup_timeseries_output_to_daily_or_monthly` or similar method that includes ALL of this code?

E.g.:
```
resilience.timeseries_output = resilience_timeseries
if ['daily', 'monthly'].include? args[:timeseries_frequency]
  resilience.timeseries_output = rollup_timeseries_output_to_daily_or_monthly(resilience.timeseries_output, ...)
end
```